Add emulation_type parameter to storage group and volume creation function

travatine opened the pull request:
Add 'emulation_type' parameter to allow creation of 'mainframe' CKD storage groups and 'CKD' volumes.
The 'deprecated' versions of 'create_' functions have not been updated as they are being removed in a future version
This fixes dell#143 .

travatine committed:
Add emulation_type parameter to storage group and volume creation function

Add 'emulation_type' parameter to allow creation of 'mainframe' CKD storage groups and 'CKD' volumes.
The 'deprecated' versions of 'create_' functions have not been updated as they are being removed in v10

travatine committed:
Merge pull request #1 from travatine/travatine-patch-1

Add emulation_type parameter to storage group and volume creation fun…

helenwalsh reviewed (changes requested):
First of all thank you so much for this submission.

All looks good but I notice that one or two places could break pep8.  These are formatting, style changes only.

$ tox -e pep

Also it may be no harm to add a unit test.  Something like 'test_create_storage_group_ckd' and 'test_modify_storage_group_ckd'  These would be added to the following file. 
There are several examples you can follow. PyU4V/PyU4V/tests/unit_tests/test_pyu4v_provisioning.py

All details can be found in https://pyu4v.readthedocs.io/en/latest/contribute.html

Thanks again
Helen

travatine committed:
Fix whitespace in provisioning.py and add 5 unit tests

New Tests:
test_create_storage_group_ckd_no_slo Create an empty CKD Storage Group
test_create_storage_group_ckd Create a CKD Storage Group with one Volume
test_add_new_vol_to_storage_group_no_name_ckd Add a CKD Volume to a SG
test_add_new_vol_to_storage_group_name_async Add CKD Volume to SG async
test_add_new_vol_to_storage_group_srdf_multihop_srdf_ckd Add a CKD Volume to a SRDF SG

travatine committed:
Correct parameters in new test method calls

Correct the parameters in :
test_create_storage_group_ckd_no_slo
test_create_storage_group_ckd
test_add_new_vol_to_storage_group_no_name_ckd
test_add_new_ckd_vol_to_storage_group_name_async
test_add_new_vol_to_storage_group_srdf_multihop_srdf_ckd

Update expected result in to include added default paremeter 'FBA':
test_create_non_empty_storage_group
test_create_non_empty_storage_group_mobility_id
test_create_empty_storage_group

travatine committed:
Add initial Split and CU Image functions:

get_split_list    Get list of FICON splits from array
get_split     Get details of a specified FICON split
get_cu_image_list     Get list of CU Image SSIDs within a FICON Split
get_cu_image   Get details of a specified CU Image
create_cu_image  Creates a new CU image under the specified split
get_cu_image_volumes    Get list of Volumes from a specified CU Image
get_cu_image_volume   Get details of a volume mapped to a  CU Image
modify_cu_image  Modify a CU - (un)assign alias / (un)map volumes

TODO: Need to make some tests...

rawstorage reviewed (approved):
Approved for merge

travatine committed:
Add unit tests for cu image functions

travatine committed:
Fix typo after adding cu functions

need to remove .conn from calls
